plugin-name/includes/Utils.php

```diff
@@ -1,19 +1,42 @@
 <?php
 /**
- * Utility Functions
+ * Utility methods
  *
- * Functions used across the plugin that are just useful.
- *
- * @package  Plugin_Name
- * @version  1.0.0
+ * @class       Utils
+ * @version     1.0.0
+ * @package     Plugin_Name/Classes/
  */
 
 namespace Plugin_Name;
 
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
 /**
- * Template Class.
+ * Utils class
  */
-class Utils {
+final class Utils {
+
+	/**
+	 * What type of request is this?
+	 *
+	 * @param  string $type admin, ajax, cron or frontend.
+	 * @return bool
+	 */
+	public static function is_request( $type ) {
+
+		switch ( $type ) {
+			case 'admin':
+				return is_admin();
+			case 'ajax':
+				return defined( 'DOING_AJAX' );
+			case 'cron':
+				return defined( 'DOING_CRON' );
+			case 'frontend':
+				return ( ! is_admin() || defined( 'DOING_AJAX' ) ) && ! defined( 'DOING_CRON' );
```

Comment:
```suggestion
				return defined( 'DOING_AJAX' ) && DOING_AJAX;
			case 'cron':
				return defined( 'DOING_CRON' ) && DOING_CRON;
			case 'frontend':
				return ( ! is_admin() || ( defined( 'DOING_AJAX' ) && DOING_AJAX ) ) && ( ! defined( 'DOING_CRON' ) || ! DOING_CRON );
```
not sure if enough to only check if the constant has been defined. If a third party added the following:
`defined( 'DOING_AJAX' ) || define( 'DOING_AJAX', false );`
i believe `defined( 'DOING_AJAX' )` would return `true`.